File: 프로그래머스/1/340213. ［PCCP 기출문제］ 1번 ／ 동영상 재생기/［PCCP 기출문제］ 1번 ／ 동영상 재생기.py
def to_second(time):
    minute, second = map(int, time.split(':'))
    return minute * 60 + second

def to_minute(time):
    minute = time // 60
    second = time % 60
    if minute < 10:
        minute = f'0{minute}'
    if second < 10:
        second = f'0{second}'
        
    return f'{minute}:{second}'
    
def solution(video_len, pos, op_start, op_end, commands):
    video_len_second = to_second(video_len)
    pos_second = to_second(pos)
    op_start_second = to_second(op_start)
    op_end_second = to_second(op_end)
    
    if pos_second < 10:
        pos_second = 0
    elif pos_second > video_len_second-10:
        pos_second = video_len_second
    if op_start_second <= pos_second <= op_end_second:
        pos_second = op_end_second
        
    for command in commands:
        if command == 'next':
            pos_second += 10
        elif command == 'prev':
            pos_second -= 10
            
        if pos_second < 10:
            pos_second = 0
        elif pos_second > video_len_second-10:
            pos_second = video_len_second
        if op_start_second <= pos_second <= op_end_second:
            pos_second = op_end_second
            
    return to_minute(pos_second)

File: 프로그래머스/1/340213. ［PCCP 기출문제］ 1번 ／ 동영상 재생기/［PCCP 기출문제］ 1번 ／ 동영상 재생기.java
class Solution {
    public int to_second(String time) {
        String[] splited_time = time.split(":");
        return Integer.parseInt(splited_time[0]) * 60 + Integer.parseInt(splited_time[1]);
    }
    
    public String to_minute(int time) {
        int minute = time / 60;
        int second = time % 60;
        return String.format("%02d:%02d", minute, second);
    }
    
    public String solution(String video_len, String pos, String op_start, String op_end, String[] commands) {
        int video_len_second = to_second(video_len);
        int pos_second = to_second(pos);
        int op_start_second = to_second(op_start);
        int op_end_second = to_second(op_end);
        
        if (pos_second < 10) {pos_second = 0;}
        else if (pos_second > video_len_second-10) {pos_second = video_len_second;}
        if (op_start_second <= pos_second && pos_second <= op_end_second) {pos_second = op_end_second;}
        
        
        for (String command: commands) {            
            if (command.equals("next")) {pos_second += 10;}
            else if (command.equals("prev")) {pos_second -= 10;}

            if (pos_second < 10) {pos_second = 0;}
            else if (pos_second > video_len_second-10) {pos_second = video_len_second;}
            if (op_start_second <= pos_second && pos_second <= op_end_second) {pos_second = op_end_second;}
        }
        
        return to_minute(pos_second);
    }
}